src/core/src/UsdUtil.cpp

```diff
@@ -1,63 +1,100 @@
 #include "cesium/omniverse/UsdUtil.h"
 
+#include "CesiumUsdSchemas/cartographicPolygon.h"
+
+#include "cesium/omniverse/AssetRegistry.h"
 #include "cesium/omniverse/Context.h"
-#include "cesium/omniverse/GeospatialUtil.h"
-#include "cesium/omniverse/Tokens.h"
+#include "cesium/omniverse/MathUtil.h"
+#include "cesium/omniverse/OmniData.h"
+#include "cesium/omniverse/OmniGeoreference.h"
+#include "cesium/omniverse/UsdTokens.h"
 #include "cesium/omniverse/Viewport.h"
 
+#include <Cesium3DTilesSelection/ViewState.h>
 #include <CesiumGeometry/Transforms.h>
 #include <CesiumGeospatial/Cartographic.h>
+#include <CesiumGeospatial/Ellipsoid.h>
 #include <CesiumGeospatial/GlobeAnchor.h>
 #include <CesiumGeospatial/GlobeTransforms.h>
+#include <CesiumGeospatial/LocalHorizontalCoordinateSystem.h>
 #include <CesiumUsdSchemas/cartographicPolygon.h>
+#include <CesiumUsdSchemas/data.h>
+#include <CesiumUsdSchemas/georeference.h>
+#include <CesiumUsdSchemas/globeAnchorAPI.h>
+#include <CesiumUsdSchemas/ionImagery.h>
+#include <CesiumUsdSchemas/ionServer.h>
+#include <CesiumUsdSchemas/polygonImagery.h>
+#include <CesiumUsdSchemas/session.h>
+#include <CesiumUsdSchemas/tileset.h>
 #include <glm/gtc/matrix_access.hpp>
 #include <glm/gtc/matrix_inverse.hpp>
-#include <glm/gtx/matrix_decompose.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtc/quaternion.hpp>
 #include <omni/ui/ImageProvider/DynamicTextureProvider.h>
 #include <pxr/base/gf/rotation.h>
 #include <pxr/usd/sdf/primSpec.h>
 #include <pxr/usd/usd/prim.h>
 #include <pxr/usd/usd/stage.h>
 #include <pxr/usd/usd/timeCode.h>
-#include <pxr/usd/usdGeom/basisCurves.h>
 #include <pxr/usd/usdGeom/metrics.h>
 #include <pxr/usd/usdGeom/xform.h>
 #include <pxr/usd/usdGeom/xformCommonAPI.h>
 #include <pxr/usd/usdShade/material.h>
 #include <pxr/usd/usdShade/shader.h>
 #include <spdlog/fmt/fmt.h>
 
-#include <regex>
+#include <cctype>
 
 namespace cesium::omniverse::UsdUtil {
 
-pxr::UsdStageRefPtr getUsdStage() {
-    return Context::instance().getStage();
-}
+namespace {
 
-long getUsdStageId() {
-    return Context::instance().getStageId();
-}
+class ScopedEdit {
```

Comment:
I wonder if class should live in the header file. It seems generally useful, so seeing it if perusing the headers sounds useful. Outside of some wrappers in FabricUtil.cpp, we don't declare other classes in .cpp files.

Reply:
Yeah it could be useful outside of here. I moved it to its own file.